sql-plugin/src/main/scala/com/nvidia/spark/rapids/GpuTransitionOverrides.scala

```diff
@@ -428,6 +430,25 @@ class GpuTransitionOverrides extends Rule[SparkPlan] {
     plan.children.foreach(assertIsOnTheGpu(_, conf))
   }
 
+  /**
+   * This is intended for testing only and this only supports looking for an exec once.
+   */
+  private def validateExecsInGpuPlan(plan: SparkPlan, conf: RapidsConf): Unit = {
+    val validateExecs = conf.validateExecsInGpuPlan.toSet
+    if (validateExecs.nonEmpty) {
+      def planContainsInstanceOf(plan: SparkPlan): Boolean = {
+        validateExecs.contains(plan.getClass.getSimpleName)
+      }
+      // to set to make uniq execs
+      val execsFound = GpuTransitionOverrides.findOperators(plan, planContainsInstanceOf).toSet
+      val execsNotFound = validateExecs.diff(execsFound.map(_.getClass().getSimpleName))
+      if (execsNotFound.nonEmpty) {
+        throw new IllegalArgumentException(
+          s"Plan ${plan.toString()} does not contain the following execs: $execsNotFound")
+      }
```

Comment:
nit: we can use the syntax sugar `require` to produce this IllegalArgumentException
```suggestion
      require(execsNotFound.nonEmpty, s"Plan ${plan.toString()} does not contain the following execs: $execsNotFound")
```

Reply:
updated